include/swift/AST/DiagnosticsSema.def

```diff
@@ -3337,6 +3337,33 @@ ERROR(diff_params_clause_param_not_differentiable,none,
       "can only differentiate with respect to parameters that conform to "
       "'Differentiable', but %0 does not conform to 'Differentiable'", (Type))
 
+// completionHanderAsync attribute
+ERROR(attr_completion_handler_async_handler_not_func,none,
+      "'%0' should be attached to non-async completion-handler function",
+      (DeclAttribute))
+
+NOTE(note_attr_function_declared_async,none,
+     "function declared async", ())
+
+NOTE(note_attr_completion_function_must_return_void,none,
+     "completion handler must return 'Void'", ())
+
+NOTE(note_attr_completion_handler_async_type_is_not_function,none,
+     "%0 is not a function type", (Type))
+
+NOTE(note_attr_completion_handler_async_handler_attr_req,none,
+     "completion handler must%select{not|}0 be '@%1'",
```

Comment:
`{ not|}` (added space), as a side note there's no test for the not case.

Reply:
Fixed